qt/aqt/theme.py

```diff
@@ -81,8 +75,43 @@ def set_night_mode(self, val: bool) -> None:
 
     night_mode = property(get_night_mode, set_night_mode)
 
+    def svg(self, path: str, color: tuple[str, str] = colors.FG) -> str:
+        "Create SVG copy with specified fill color."
+        from aqt.utils import aqt_data_folder
+
+        cache = self._svg_cache
+
+        if m := re.match(r"(?:icons:)(.+)(?:\.svg)", path):
+            name = m.group(1)
+        else:
+            return path
+
+        key = f"{name}-{color[1 if self.night_mode else 0]}"
+
+        existing_path = cache.get(key)
+        if existing_path:
+            return existing_path
+
+        src_path = os.path.join(aqt_data_folder(), "qt", re.sub(r":", "/", path))
+        with open(src_path, "r", encoding="UTF-8") as file:
+            data = file.read()
+
+        if "fill" in data:
+            data = re.sub(r"fill=\"#.+?\"", f'fill="{self.var(color)}"', data)
+        else:
+            data = re.sub(r"<svg", f'<svg fill="{self.var(color)}" ', data, 1)
+
+        copy_path = os.path.join(
+            os.path.join(aqt_data_folder(), "qt", "icons"), f"{key}.svg"
+        )
+        # create copy with specified fill color
+        with open(copy_path, "w", encoding="UTF-8") as file:
+            file.write(data)
```

Comment:
Anki may be installed in a location where its data files are not writable, which will cause this to fail. The files will either need to be placed somewhere like a temp folder (which will introduce problems with temp file cleaners), or generated up-front. I'd lean towards the latter, as the extra files aren't going to contribute a large amount to the total bundle size, and having them generated in advance means there's no risk of impacting startup time. I'll push a commit that shows how you can do this with Bazel.

Reply:
Thanks for showing me this! I modified the script to allow an arbitrary amount of color variants for each icon.